pkg/settings/setting.go

```diff
@@ -129,12 +125,33 @@ var (
 
 	Rke2DefaultVersion = NewSetting("rke2-default-version", "")
 	K3sDefaultVersion  = NewSetting("k3s-default-version", "")
+
+	// KubeconfigGenerateToken determines whether the UI will return a generate token with kubeconfigs.
+	// If set to false the kubeconfig will contain a command to login to Rancher.
+	KubeconfigGenerateToken = NewSetting("kubeconfig-generate-token", "true")
+
+	// Deprecated: On removal use kubeconfig-default-ttl-minutes for all kubeconfigs.
+	KubeconfigTokenTTLMinutes = NewSetting("kubeconfig-token-ttl-minutes", "960") // 16 hours
+
+	// KubeconfigDefaultTokenTTLMinutes is the default time to live applied to kubeconfigs created for users.
+	KubeconfigDefaultTokenTTLMinutes = NewSetting("kubeconfig-default-ttl-minutes", "0") // 0 TTL = never expire
+
+	// AuthTokenMaxTTLMinutes is the max allowable time to live for tokens. Excluding those created for UI sessions which is controlled by AuthUserSessionTTLMinutes.
```

Comment:
Isn't `AuthUserSessionTTLMinutes` actually capped by this one ?

Reply:
Not currently, One place the session token is set is  [here](https://github.com/rancher/rancher/blob/836b43cec20f4a803f82982f21aa3a1f965275c6/pkg/auth/providers/publicapi/login.go#L109 ) without clamping to `AuthTokenMaxTTLMinutes`. I also ran an end-to-end test to confirm this behavior